plugins/woocommerce/src/Blocks/BlockTypes/ProductStockIndicator.php

```diff
@@ -78,42 +47,46 @@ protected static function getTextBasedOnStock( $is_in_stock, $is_low_stock, $low
 	 * @return string Rendered block type output.
 	 */
 	protected function render( $attributes, $content, $block ) {
-		if ( ! empty( $content ) ) {
-			parent::register_block_type_assets();
-			$this->register_chunk_translations( [ $this->block_name ] );
-			return $content;
-		}
-
 		$post_id = isset( $block->context['postId'] ) ? $block->context['postId'] : '';
 		$product = wc_get_product( $post_id );
 
 		if ( ! $product ) {
 			return '';
 		}
 
-		$is_in_stock     = $product->is_in_stock();
-		$is_on_backorder = $product->is_on_backorder();
+		$availability = $product->get_availability();
+
+		if ( empty( $availability['availability'] ) ) {
+			return '';
+		}
 
 		$low_stock_amount = $product->get_low_stock_amount();
 		$total_stock      = $product->get_stock_quantity();
-		$is_low_stock     = $low_stock_amount && $total_stock <= $low_stock_amount;
-
+		$is_low_stock       = $low_stock_amount && $total_stock <= $low_stock_amount;
 		$classes_and_styles = StyleAttributesUtils::get_classes_and_styles_by_attributes( $attributes );
 
 		$classnames  = isset( $classes_and_styles['classes'] ) ? ' ' . $classes_and_styles['classes'] . ' ' : '';
-		$classnames .= ! $is_in_stock ? ' wc-block-components-product-stock-indicator--out-of-stock ' : '';
-		$classnames .= $is_in_stock ? ' wc-block-components-product-stock-indicator--in-stock ' : '';
-		$classnames .= $is_low_stock ? ' wc-block-components-product-stock-indicator--low-stock ' : '';
-		$classnames .= $is_on_backorder ? ' wc-block-components-product-stock-indicator--available-on-backorder ' : '';
+		$classnames .= sprintf( ' wc-block-components-product-stock-indicator--%s', $availability['class'] );
+
+		$is_backorder_notification_visible = $product->is_in_stock() && $product->backorders_require_notification();
+
+		if ( empty( $content ) && $is_backorder_notification_visible ) {
```

Comment:
When `$content` is empty, we assume we are displaying a `Product Collection`.

Reply:
This change was added to modify the text `(can be back ordered)` in product collections. Currently, we present a shorter version of the text; otherwise, it would take up two lines:

| in two lines | in one line |
|--------|--------|
| ![Screenshot 2024-12-05 at 4 33 21 PM](https://github.com/user-attachments/assets/f07d0465-ca8d-4556-ab46-454434a46b64) | ![Screenshot 2024-12-05 at 4 28 36 PM](https://github.com/user-attachments/assets/0cb1ad48-bd76-48ba-9982-eac42ddeb38f) |